mem.go

```diff
@@ -168,21 +168,27 @@ func (m *Message) NumSegments() int64 {
 // Segment returns the segment with the given ID.
 func (m *Message) Segment(id SegmentID) (*Segment, error) {
 	m.mu.Lock()
```

Comment:
This potentially makes sense.  We don't have a use case where we're frequently accessing the data concurrently, but it would certainly make a difference in highly concurrent applications.

To add to that, I wouldn't be surprised if most capnp use cases are write once, read a lot.